### PR TITLE
feat(room): add invite code support

### PR DESCRIPTION
**Pull request by BlackHole1**
TODO: add test

### CODE REVIEW

File: src/v1/controller/room/updateStatus/Stopped.ts [hunk 1]
```diff
@@ -172,3 +189,29 @@ interface RequestType {
...
+    const inviteCode = await RedisService.get(RedisKey.roomInviteCodeReverse(roomUUID));
+
+    if (inviteCode === null) {
+        return;
+    }
+
+    await RedisService.client
+        .multi()
+        .expire(RedisKey.roomInviteCode(inviteCode), tenDay)
+        .expire(RedisKey.roomInviteCodeReverse(roomUUID), tenDay)
```
Comment (BlackHole1) [inline src/v1/controller/room/updateStatus/Stopped.ts:207]: Seems unnecessary?

File: src/v1/controller/room/create/Utils.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        inviteCode = roomUUID;
+    }
+
+    if (inviteCode !== roomUUID) {
+        const fiftyDays = 60 * 60 * 24 * 50;
+
+        await RedisService.client
+            .multi()
+            .set(RedisKey.roomInviteCode(inviteCode), roomUUID, "EX", fiftyDays)
+            .set(RedisKey.roomInviteCodeReverse(roomUUID), inviteCode, "EX", fiftyDays)
```
Comment (BlackHole1) [inline src/v1/controller/room/create/Utils.ts:25]: SETNX should be used

File: src/v1/controller/room/create/Utils.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            .then(data => {
+                for (let i = 0; i < data.length; i++) {
+                    const error = data[i][0];
+                    if (error !== null) {
+                        throw error;
+                    }
+                }
+            })
+            .catch(error => {
+                logger.warn("set room invite code to redis failed", parseError(error));
```
Comment (BlackHole1) [inline src/v1/controller/room/create/Utils.ts:36]: add

```ts
inviteCode = roomUUID
```

File: src/v1/controller/room/updateStatus/Stopped.ts (hunk 1 above)
Comment (BlackHole1) [reply src/v1/controller/room/updateStatus/Stopped.ts:207]: see: https://github.com/netless-io/flat-server/pull/403/files#diff-a6dad097cfc2eb9322a610ff90342c9c4cabe811673b8179627a5cda3dd4efe7R194-R195

**Review by crimx — APPROVED** [review]

**Pull request merged**